File: drand48/driver_c/generate.c
#include <stdio.h>
#include <stdlib.h>

int main( int argc, char *argv[] ) {
    if ( argc != 3 ) {
        printf("usage: %s <seed> <number of iterations>\n", argv[0]);
        return 1;
    }

    int seed           = atoi( argv[1] );
    int num_iterations = atoi( argv[2] );


    srand48(seed);

    int iteration;
    for ( iteration = 0; iteration < num_iterations; iteration++) {
        printf("%f\n", drand48());
    }

    return 0;
}

File: drand48/driver_perl/generate.pl
#!/usr/bin/env perl

use strict;
use warnings;

use Config;

# Make sure we are using drand48 as the underlying implementation
if($Config{randfunc} ne "drand48") {
    print "Perl is configured with randfunc '" . $Config{randfunc}
        . "' rather than drand48\n";
    return 1;
}

if( scalar @ARGV != 2 ) {
    print "usage: $0 <seed> <number of iterations>\n";
    return 1;
}

my $seed           = $ARGV[0];
my $num_iterations = $ARGV[1];

srand($seed);

for (my $iteration = 0; $iteration < $num_iterations; $iteration++) {
    printf("%f\n", rand());
}
